egs/librispeech/asr/simple_v1/bpe_mmi_att_transformer_train.py

```diff
@@ -0,0 +1,716 @@
+    parser.add_argument(
+        '--attention-dim',
+        type=int,
+        default=256,
+        help="Number of units in transformer attention layers.")
+    parser.add_argument(
+        '--tensorboard',
+        type=str2bool,
+        default=True,
+        help='Should various information be logged in tensorboard.'
+    )
+    parser.add_argument(
+        '--amp',
+        type=str2bool,
+        default=True,
+        help='Should we use automatic mixed precision (AMP) training.'
+    )
+    parser.add_argument(
+        '--use-ali-model',
+        type=str2bool,
+        default=True,
+        help='If true, we assume that you have run ./ctc_train.py '
+             'and you have some checkpoints inside the directory '
+             'exp-lstm-adam-ctc-musan/ .'
+             'It will use exp-lstm-adam-ctc-musan/epoch-{ali-model-epoch}.pt '
+             'as the pre-trained alignment model'
+    )
+    parser.add_argument(
+        '--ali-model-epoch',
+        type=int,
+        default=7,
+        help='If --use-ali-model is True, load '
+             'exp-lstm-adam-ctc-musan/epoch-{ali-model-epoch}.pt as the alignment model.'
+             'Used only if --use-ali-model is True.'
+    )
+    parser.add_argument(
+        '--use-pruned-intersect',
+        type=str2bool,
+        default=False,
+        help='True to use pruned intersect to compute the denominator lattice. ' \
+             'You probably want to set it to True if you have a very large LM. ' \
+             'In that case, you will get an OOM if it is False. ')
+    #  See https://github.com/k2-fsa/k2/issues/739 for more details
+    parser.add_argument(
+        '--torchscript',
+        type=str2bool,
+        default=False,
+        help='Should we convert the model to TorchScript before starting training.'
+    )
+    parser.add_argument(
+        '--torchscript-epoch',
+        type=int,
+        default=-1,
+        help='After which epoch should we start storing models with TorchScript,'
+             'so that they can be simply loaded with torch.jit.load(). '
+             '-1 disables this option.'
+    )
+    return parser
+
+
+def run(rank, world_size, args):
+    '''
+    Args:
+      rank:
+        It is a value between 0 and `world_size-1`, which is
+        passed automatically by `mp.spawn()` in :func:`main`.
+        The node with rank 0 is responsible for saving checkpoint.
+      world_size:
+        Number of GPUs for DDP training.
+      args:
+        The return value of get_parser().parse_args()
+    '''
+    model_type = args.model_type
+    start_epoch = args.start_epoch
+    num_epochs = args.num_epochs
+    accum_grad = args.accum_grad
+    den_scale = args.den_scale
+    att_rate = args.att_rate
+    use_pruned_intersect = args.use_pruned_intersect
+
+    fix_random_seed(42)
+    setup_dist(rank, world_size, args.master_port)
+
+    exp_dir = Path('exp-bpe-' + model_type + '-mmi-att-sa-vgg-normlayer')
```

Comment:
This file is copied from mmi_att_transformer_train.py, with only this line being modified.